### PR TITLE
PT-1432 | Implement new occurrence form with HDS inputs

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

Start using HDS components to set date and time for occurrences.

- Modify form to work with date and time components
- Add new checkbox for multiday occurrences
- Add confirmation modal when trying to add ocurrences with same start and end time

### Closes :no_good_woman:
**[PT-1432](https://helsinkisolutionoffice.atlassian.net/browse/PT-1432):** 

### Manual testing :construction_worker_man:

1. Create new event
2. Add occurrences with new form in occurrences page (2nd step)
3. Add multiday occurrences
4. Try to add occurrence with same start and end time twice

## Screenshots :camera_flash:

Default form 
Screenshot 2022-01-13 at 11 59 36

Multiday form
Screenshot 2022-01-13 at 11 59 45

Confirmation modal
Screenshot 2022-01-13 at 11 59 55

## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/occurrence/occurrencesFormPart/OccurrencesFormPart.tsx [hunk 1]
```diff
@@ -181,6 +201,24 @@ const OccurrencesForm: React.FC<{
     }
   };
 
+  // TODO: what to do is this is called when going to publish page and confirmation modal is opened?
```
Comment (Jokinen) [inline src/domain/occurrence/occurrencesFormPart/OccurrencesFormPart.tsx:204]: Is this a TODO you intended to resolve in this PR?

**Review by Jokinen — APPROVED** [review]
Looks good to me.

Very minor comments on the UI:
Screenshot 2022-01-17 at 15 53 42

Language select can't fit value on an average laptop screen.

Quite extensive changes when it comes to the code, but couldn't spot anything I wanted to point out.

File: src/domain/occurrence/occurrencesFormPart/utils.ts [hunk 2]
```diff
@@ -199,6 +207,28 @@ export const deleteOccurrenceFromCache = ({
   });
 };
 
+export const getOccurrencerWithSameDateAlreadyExists = (
```
Comment (Sartonon) [inline src/domain/occurrence/occurrencesFormPart/utils.ts:210]: typo (note for myself)

Comment (nikomakela) [reply src/domain/occurrence/occurrencesFormPart/utils.ts:210]: - [ ] fix a typo ;)

File: src/domain/app/i18n/sv.json [hunk 3]
```diff
@@ -544,9 +544,10 @@
     "infoText2": "Registrering för denna evenemangtid är möjlig efter <strong>{{date}}</strong> och stängs <strong>{{count}}</strong> dag före evenemangstiden. Du kan ändra dessa inställningar i händelsesinformationen.",
     "infoText2_plural": "Registrering för denna evenemangtid är möjlig efter <strong>{{date}}</strong> och stängs <strong>{{count}}</strong> dagar före evenemangstiden. Du kan ändra dessa inställningar i händelsesinformationen.",
     "helpTextFirstOccurrence": "Fyll i Informationen för den första evenemang",
-    "labelDate": "Datum",
-    "labelStartsAt": "Börjar kl",
-    "labelEndsAt": "Slutar kl",
+    "labelStartDate": "Datum",
+    "labelStartTime": "Börjar kl",
+    "labelEndDate": "SV: End date",
```
Comment (nikomakela) [inline src/domain/app/i18n/sv.json:549]: Needs a translation

File: src/domain/app/i18n/sv.json [hunk 4]
```diff
@@ -562,7 +563,24 @@
...
+      "labelStartMinutes": "SV: Alkuajan minuutit",
+      "labelEndHours": "SV: Loppuajan tunnit",
+      "labelEndMinutes": "SV: Loppuajan minuutit"
+    },
+    "duplicateOccurrenceModal": {
+      "title": "SV: Huomio",
+      "text": "SV: Tapahtumaan on jo lisätty tapahtuma-aika samalla alku- ja loppuajalla. Haluatko varmasti lisätä tämän tapahtuma-ajan?",
+      "buttonAddOccurrences": "SV: Lisää tapahtuma-aika",
+      "buttonCancelAddOccurrence": "SV: Peruuta"
+    }
```
Comment (nikomakela) [inline src/domain/app/i18n/sv.json:583]: Translations needed

File: src/domain/app/i18n/sv.json [hunk 5]
```diff
@@ -582,12 +600,14 @@
         "positiveNumber": "Detta fält måste vara ett giltigt nummer"
       },
       "date": {
+        "invalid": "SV: Use form D.M.YYYY",
```
Comment (nikomakela) [inline src/domain/app/i18n/sv.json:603]: translations needed

File: src/domain/app/i18n/sv.json [hunk 6]
```diff
@@ -582,12 +600,14 @@
...
         "required": "Obligatorikst att fylla detta fält",
         "mustNotInThePast": "Detta datum kan inte vara i förr",
         "min": "Detta datum måste vara tidigast {{min}}",
         "max": "Detta datum måste vara senast {{max}}"
       },
       "time": {
+        "invalid": "SV: Time is invalid",
```
Comment (nikomakela) [inline src/domain/app/i18n/sv.json:610]: translations needed

**Review by nikomakela — CHANGES_REQUESTED** [review]
Needs some Swedish translations before can be merged

**Comment by nikomakela** [comment]
I had some serious issues with the datepicker input when I tested it. I could not set a date there - at all. 
image
Browser is the Chrome, but it's in English - could it make the difference?

**Pull request merged**